pkg/catalog/outbound_traffic_policies.go

```diff
@@ -182,14 +187,18 @@ func (mc *MeshCatalog) buildOutboundPolicies(sourceServiceIdentity identity.Serv
 		// Do not build an outbound policy if the destination service is an apex service in a traffic target
 		// this will be handled while building policies from traffic split (with the backend services as weighted clusters)
 		if !mc.isTrafficSplitApexService(destService) {
-			hostnames, err := mc.getServiceHostnames(destService, source.Namespace == destService.Namespace)
+			locality := service.LocalCluster
+			if destService.Namespace == source.Namespace {
+				locality = service.LocalNS
+			}
```

Comment:
I am not sure I understand the logic here, the source and destination can be in the same namespace but on different clusters. I don't think having locality represent the cluster and namespace is the right way forward, they need to be separate.

Example: bookbuyer and bookstore can be the the same ns but can be on two different clusters.

Reply:
We think of this as concentric circles. A namespace within a namespace is reachable without specifying the namespace subdomain. A service within the same cluster is reachable without specifying the FQDN. Outside of the cluster, the FQDN is required, *even* if they are in the same namespace

Reply:
> We think of this as concentric circles. A namespace within a namespace is reachable without specifying the namespace subdomain. A service within the same cluster is reachable without specifying the FQDN. Outside of the cluster, the FQDN is required, _even_ if they are in the same namespace

@steeling what you mention above makes sense, though clients are allowed to use full or partial FQDNs depending on where the service is located. Ex. client in ns `ns1` should be able to access service `foo` in `ns1` using `foo.ns1.svc.cluster.local`, `foo.ns1.svc.cluster`, `foo.ns1.svc`, `foo.ns1`, or just `foo`. I hope we aren't breaking this use case.

Reply:
correct we are maintaining this. I think I misused the term FQDN in referring strictly full FQDN vs partial, but yes we are keeping this case.

Reply:
to clarify, any partial FQDN will resolve to local, whereas to access the remote it needs the full FQDN

Reply:
makes sense, I assume that the logic to build the FQDN will change then once its identified to be a local or remote cluster ?

Reply:
Yup! That will come in a PR early next week

Reply:
got it, thanks for clarifying